src/sagemaker/jumpstart/model.py

```diff
@@ -402,6 +403,42 @@ def retrieve_example_payload(self) -> JumpStartSerializablePayload:
             sagemaker_session=self.sagemaker_session,
         )
 
+    @classmethod
+    def attach(
+        cls,
+        endpoint_name: str,
+        inference_component_name: Optional[str] = None,
+        model_id: Optional[str] = None,
+        model_version: Optional[str] = None,
+        sagemaker_session=DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+    ) -> "JumpStartModel":
+        """Attaches a JumpStartModel object to an existing SageMaker Endpoint.
+
+        The model id, version (and inference component name) can be inferred from the tags.
+        """
+
+        inferred_model_id, inferred_model_version, inferred_inference_component_name = (
+            get_model_id_version_from_endpoint(
+                endpoint_name=endpoint_name,
+                inference_component_name=inference_component_name,
+                sagemaker_session=sagemaker_session,
+            )
+        )
```

Comment:
non-blocking: how about skipping this call if a/ `model_id`, `model_version` and `inference_component_name` are specified or b/ `model_id` and `model_version` are specified and the EP is model-based (not IC-based)?

Reply:
Good idea. This increases latency unnecessarily

Reply:
You don't know a priori if the endpoint is IC based, and to figure that out would require an API call, so I will skip this check if all 3 fields are provided